minio/sse.py

```diff
@@ -56,7 +56,7 @@ def __init__(self, key):
                 "SSE-C keys need to be 256 bit base64 encoded",
             )
         b64key = base64.b64encode(key).decode()
-        md5 = hashlib.md5()
+        md5 = hashlib.new("md5", usedforsecurity=False)
         md5.update(key)
         md5key = base64.b64encode(md5.digest()).decode()
```

Comment:
Just remove these lines and use `md5key = helpers.md5sum_hash(key)`.

Reply:
i can not commit this due to a circular import.

```patch
diff --git a/minio/sse.py b/minio/sse.py
index 5b6d0c3..5051c20 100644
--- a/minio/sse.py
+++ b/minio/sse.py
@@ -25,10 +25,11 @@ This module contains core API parsers.
 
 """
 import base64
-import hashlib
 import json
 from abc import ABCMeta, abstractmethod
 
+from .helpers import md5sum_hash
+
 
 class Sse:
     """Server-side encryption base class."""
@@ -56,9 +57,7 @@ class SseCustomerKey(Sse):
                 "SSE-C keys need to be 256 bit base64 encoded",
             )
         b64key = base64.b64encode(key).decode()
-        md5 = hashlib.new("md5", usedforsecurity=False)
-        md5.update(key)
-        md5key = base64.b64encode(md5.digest()).decode()
+        md5key = md5sum_hash(key)
         self._headers = {
             "X-Amz-Server-Side-Encryption-Customer-Algorithm": "AES256",
             "X-Amz-Server-Side-Encryption-Customer-Key": b64key,
```


it is about `from .sse import Sse, SseCustomerKey` in `helpers.py`

Reply:
Below is the fix
```diff
diff --git a/minio/sse.py b/minio/sse.py
index 8188c79..eef5019 100644
--- a/minio/sse.py
+++ b/minio/sse.py
@@ -25,7 +25,6 @@ This module contains core API parsers.
 
 """
 import base64
-import hashlib
 import json
 from abc import ABCMeta, abstractmethod
 
@@ -56,9 +55,9 @@ class SseCustomerKey(Sse):
                 "SSE-C keys need to be 256 bit base64 encoded",
             )
         b64key = base64.b64encode(key).decode()
-        md5 = hashlib.md5()
-        md5.update(key)
-        md5key = base64.b64encode(md5.digest()).decode()
+        from .helpers import \
+            md5sum_hash  # pylint: disable=import-outside-toplevel
+        md5key = md5sum_hash(key)
         self._headers = {
             "X-Amz-Server-Side-Encryption-Customer-Algorithm": "AES256",
             "X-Amz-Server-Side-Encryption-Customer-Key": b64key,
```